### PR TITLE
Cloak pragma and fix tests for 2.1.0-dev

### LINKED ISSUE #1726: Avoid @pragma in documentation
The VM is adding `@pragma` annotations to the SDK source (https://dart-review.googlesource.com/c/sdk/+/65545). These must not show up in the API documentation. Pragma annotations are, by definition, only intended for specific tools, not for users, and there is only negative value in exposing them.

### PR DESCRIPTION
**Pull request by jcollins-g**
Fixes #1726 and tests for 2.1.0.

### CODE REVIEW

**Comment by jcollins-g** [comment]
Additionally, drops the tuple dependency from the test package to eliminate confusion (we incorporated tuple into dartdoc in #1730).

File: lib/src/model.dart [hunk 1]
```diff
@@ -2729,16 +2729,42 @@ abstract class ModelElement extends Canonicalization
...
-          .findCanonicalModelElementFor(a.element?.enclosingElement);
-      if (me != null)
-        annotation = annotation.replaceFirst(me.name, me.linkedName);
-      return annotation;
-    }).toList(growable: false);
+      Element annotationElement = a.element;
+
+      ClassElement annotationClassElement;
+      if (annotationElement is ExecutableElement) {
+        annotationElement = (annotationElement as ExecutableElement)
```
Comment (devoncarew) [inline lib/src/model.dart]: I'm surprised that type promotion doesn't just allow you to type `annotationElement` w/o the cast.

Comment (jcollins-g) [reply lib/src/model.dart]: It might be because of the assignment of annotationElement to a different type, but yes it doesn't promote the type here.

Comment (lrhn) [reply lib/src/model.dart]: It doesn't promote because you are assigning to the variable inside the guarded block.

File: lib/src/model.dart [hunk 2]
```diff
@@ -4524,6 +4551,18 @@ class PackageGraph {
...
+
+  /// Returns the set of [Class] objects that are similar to pragma
+  /// in that we should never count them as documentable annotations.
+  Set<Class> get invisibleAnnotations {
+    if (_invisibleAnnotations == null) {
+      _invisibleAnnotations = new Set();
+      _invisibleAnnotations.add(specialClasses[SpecialClass.pragma]);
+    }
+    return _invisibleAnnotations;
+  }
```
Comment (lrhn) [inline lib/src/model.dart]: A matter of preferences, but this entire body can be written as:

  => _invisibleAnnotations ??= Set()..add(specialClasses[SpecialClass.pragma]);

(Code golf FTW :innocent:)

**Pull request merged**